charts/common/templates/_names.tpl

```diff
@@ -0,0 +1,62 @@
+{{/* vim: set filetype=mustache: */}}
+
+{{/*
+Get postgres host name in the following order of precendence: Chart specific postgresql > Chart external Postgresql > Global postgresql, Arguments:
+1) postgresql object
+2) external postgresql object
+3) Release object
+4) Chart object
+*/}}
+{{- define "common.postgres-host" -}}
+{{- $postgresql := index . 0 -}}
+{{- $externalPostgresql := index . 1 -}}
+{{- $release := index . 2 -}}
+{{- $chart := index . 3 -}}
+{{- if $postgresql.enabled -}}
+    {{- printf "%s-%s-postgresql.%s.svc.cluster.local" $release.Name $chart.Name $release.Namespace -}}
+{{- else if $externalPostgresql.enabled -}}
+    {{- $externalPostgresql.address -}}
+{{- else -}}
+    {{- printf "%s-postgresql.%s.svc.cluster.local" $release.Name $release.Namespace -}}
+{{- end -}}
+{{- end -}}
+
+
+{{/*
+Get postgres user name in the following order of precendence: Chart specific postgresql > Chart external Postgresql > Global postgresql, Arguments:
+1) postgresql object
+2) external postgresql object
+3) global object
+*/}}
+{{- define "common.postgres-username" -}}
+{{- $postgresql := index . 0 -}}
+{{- $externalPostgresql := index . 1 -}}
+{{- $global := index . 2 -}}
+    {{- if $postgresql.enabled -}}
+        {{- $postgresql.postgresqlUsername -}}
+    {{- else if $externalPostgresql.enabled -}}
+        {{- $externalPostgresql.username -}}
+    {{- else -}}
+        {{- $global.postgresqlUsername -}}
+    {{- end -}}
+{{- end -}}
+
+
+{{/*
+Get postgres database name in the following order of precendence: Chart specific postgresql > Chart external Postgresql > Global postgresql, Arguments:
+1) postgresql object
+2) external postgresql object
+3) global object
+*/}}
+{{- define "common.postgres-database" -}}
+{{- $postgresql := index . 0 -}}
+{{- $externalPostgresql := index . 1 -}}
+{{- $global := index . 2 -}}
+    {{- if $postgresql.enabled -}}
+        {{- $postgresql.postgresqlDatabase -}}
+    {{- else if $externalPostgresql.enabled -}}
+        {{- $externalPostgresql.database -}}
+    {{- else -}}
+        {{- $global.mlp.postgresqlDatabase -}}
```

Comment:
`mlp` - typo?

Reply:
usually, there are pairs of user-database created to manage access. I don't see a benefit in specifying mlp database in global instead in mlp values

Reply:
it was a typo. but tests passed for the case of mlp.  let me explain what's happening. 

We create application-specific database schemas in the init script of the caraml umbrella chart. [ref](https://github.com/caraml-dev/helm-charts/blob/main/charts/caraml/values.yaml#L235) 

And add the created schema name under the global variables under its key so that it can connect to the corresponding database. ideally here, we need to get the object for the app (mlp in this case) and fetch the database from it. Will make the change to do so.

Reply:
Updated the same.